src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -1116,20 +1120,24 @@ public static URL checkEndPoint(String url) throws FormValidation {
     }
 
     @CheckForNull
-    private static SSHUserPrivateKey getSshCredential(@NonNull String id, ItemGroup context){
-        SSHUserPrivateKey credential = CredentialsMatchers.firstOrNull(
-                CredentialsProvider.lookupCredentials(
-                        SSHUserPrivateKey.class, // (1)
-                        context,
-                        null,
-                        Collections.emptyList()),
-                CredentialsMatchers.withId(id));
-
-        if (credential == null) {
-            LOGGER.log(Level.WARNING, "EC2 Plugin could not find the specified credentials ({0}) in the Jenkins Global Credentials Store, EC2 Plugin for cloud must be manually reconfigured", new String[]{id});
-        }
+    private static SSHUserPrivateKey getSshCredential(String id, ItemGroup context){
+
+        if (!id.isEmpty()) {
```

Comment:
So is `id` still supposed to be `@NonNull`, as implied by the NPE here otherwise?

Reply:
yes, let me fix that